tests/network-tests/src/fixtures/council/ElectCouncilFixture.ts

```diff
@@ -7,14 +7,38 @@ import { assert } from 'chai'
 import { createType, registry } from '@joystream/types'
 
 export class ElectCouncilFixture extends BaseQueryNodeFixture {
+  protected async createCandidates(numberOfCandidates: number, electionNumber: number): Promise<string[]> {
+    const addresses: string[] = []
+    for (let i = 0; i < numberOfCandidates; ++i) {
+      addresses.push(this.api.createCustomKeyPair('//Candidate//' + electionNumber + '//' + i).address)
+    }
+    await Promise.all(addresses.map((a) => this.api.treasuryTransferBalance(a, this.api.existentialDeposit())))
+    return addresses
+  }
+
+  // needed since the first election test usually starts in the Announcing period
+  // meanwhile subsequent elections tests start in the Idle period
+  protected async getUpcomingAnnouncingPeriod(): Promise<number> {
+    const stageUpdate = await this.api.query.council.stage()
+    const stage = stageUpdate.stage
+    const announcementPeriodId = await this.api.query.council.announcementPeriodNr()
+    if (stage.isAnnouncing) {
+      return announcementPeriodId.toNumber()
+    } else {
+      return announcementPeriodId.toNumber() + 1
+    }
```

Comment:
This will not always work, because:
- It doesn't take into account the "block reserve" (ie. whether the announcing period has less than `blockReserve` blocks left)
- There are a few more actions ran after this check, for example, the `BuyMembershipHappyCaseFixture`. After those actions are finished, the stage may already be different (if the announcing period was close to an end).

One way to solve this would be to update the controller accounts of council members after they are already elected. I don't think you even need to include the election cycle number in this case.

Reply:
That might be best., that's how we do it for the working group workers -> https://github.com/Joystream/joystream/blob/997c5ae929b7422486593c3f0054154eef5f5cd2/tests/network-tests/src/Api.ts#L846

Reply:
There's no way to change a controller account for a member or councilor once set

Reply:
You can do it via `update_accounts` extrinsic in the membership module

Reply:
😃  good catch I have added the suggested fix + assertion